db/model/collector.js

```diff
@@ -235,9 +236,38 @@ module.exports = function collector(seq, dataTypes) {
 
     Collector.addScope('running', {
       where: {
-        status: constants.collectorStatuses.Running,
+        status: collectorStatus.Running,
       },
     });
+
+    Collector.addScope('currentGenerators', {
+      include: [
+        {
+          model: models.Generator.scope('embed'),
+          as: 'currentGenerators',
+        },
+      ],
+    });
+
+    Collector.addScope('possibleGenerators', {
+      include: [
+        {
+          model: models.Generator.scope('embed'),
+          as: 'possibleGenerators',
+          through: { attributes: [] },
+        },
+      ],
+    });
+
+    Collector.addScope('defaultScope',
```

Comment:
What's happening here is: We are creating a scoped model using the three already-defined scopes. This creates a new scope object and sets it as the "_scope" attribute on the model. We then access that attribute and pass it in to addScope().

This works because the _scope attribute is just a normal scope object, of the same format as any other scope that can be passed in to addScope. For example, if you use a model with the default scope, _scope is the exact object that was passed in to addScope().

This is necessary because Sequelize doesn't provide a way to combine scopes directly (the only place this happens is in Model.scope()), and the scope-combining logic is non-trivial. It's a bit of a hack, but it works, and I think it's much better than replicating the scope definitions.

The details of how this works are a little complicated, but at-a-glance I think it's clear that what we're doing is "combine these scopes". I feel like explaining what's happening in a comment would actually make it less clear.

Or am I wrong about the purpose being clear at-a-glance? Maybe I could just add a basic comment saying that that's what this does? Or extract this to a function called combineScopes()?

Reply:
The 'combining the scopes' part was clear when I looked at it, but _scope attribute part was not that clear enough - I like the idea of extracting it to a function which can be used by generator model as well. And we can add a basic comment for that function as well - what do you think?

Reply:
ok, I will try that.

Reply:
updated.